pkg/db/src/types/orders.ts

```diff
@@ -1,7 +1,40 @@
-import { DatabaseInterface as BaseDatabaseInterface } from "./misc";
+/* eslint-disable @typescript-eslint/ban-types */
+
+import { DatabaseInterface as BaseDatabaseInterface, CouchDocument } from "./misc";
 
 export type OrdersDatabaseInterface = BaseDatabaseInterface;
 
 export interface NewOrdersDatabase {
 	(db: PouchDB.Database): OrdersDatabaseInterface;
 }
+
+/**
+ * A (standardized) customer order interface:
+ * * standard data structure
+ * * standard method interface
+ */
+export type CustomerOrderInterface<A extends Record<string, any> = {}> = CustomerOrderProto<A> & CustomerOrderData<A>;
+
+/**
+ * A standardized interface (interface of methods) for a customer order.
+ * Different implementations might vary, but should always extend this interface.
+ */
+export interface CustomerOrderProto<A extends Record<string, any> = {}> {
+	create: () => Promise<CustomerOrderInterface<A>>;
+}
+
+export interface CustomerOrderBook { isbn: string; status: string };
+/**
+ * Standardized data that should be present in any customer order
+ * (different implementations might differ, but should extend this structure)
+ */
+export type CustomerOrderData<A extends Record<string, any> = {}> = CouchDocument<
+	{
+		updatedAt: string | null;
```

Comment:
I came here to check how this compared against my mock type for order table data - just a note that I have also included `name`